modules/core/src/main/java/org/apache/ignite/internal/client/impl/GridClientAbstractProjection.java

```diff
@@ -365,6 +365,44 @@ private GridClientNode balancedNode(@Nullable final GridClientNode exclude) thro
         return balancer.balancedNode(prjNodes);
     }
 
+    /**
+     * Creates exclude nodes filter. The node must support REST, not be same as {@code excludeNode} and not be excluded
+     * by custom {@code excludeNodesPred}.
+     *
+     * @param excludeNode Node to exclude (optional).
+     * @param excludeNodesPred Custom filter (optional).
+     * @return Filter.
+     */
+    private GridClientPredicate<GridClientNode> excludeFilter(
+        @Nullable GridClientNode excludeNode,
+        @Nullable GridClientPredicate<GridClientNode> excludeNodesPred
+    ) {
+        return new GridClientPredicate<GridClientNode>() {
+            @Override public boolean apply(GridClientNode n) {
+                if (excludeNode != null && excludeNode.equals(n))
+                    return false;
+
+                if (excludeNodesPred != null && excludeNodesPred.apply(n))
+                    return false;
+
+                return restAvailable(n, client.cfg.getProtocol());
+            }
+
+            @Override public String toString() {
+                StringBuilder sb = new StringBuilder("Filter nodes with available REST");
```

Comment:
its not yours, but i found it non informative toString request if excludeNode and excludeNodesPred are nulls. It`s up to you fix it or not.

Reply:
It's an internal client for the internal purposes)